src/lib/xmlparse.rs

```diff
@@ -1078,87 +1075,33 @@ pub const EXPAND_SPARE: c_int = 24;
 
 pub const INIT_SCAFFOLD_ELEMENTS: c_int = 32;
 
+fn malloc_layout(size: usize) -> Layout {
+    Layout::from_size_align(size, 1)
+        .expect("failed to create Layout")
+}
+
+fn malloc_array_layout<T>(n: usize) -> Layout {
+    Layout::array::<T>(n as usize)
+        .expect("failed to create array Layout")
+}
+
 macro_rules! MALLOC {
-    ($parser:expr, $size:expr $(,)?) => {
-        (*$parser)
-            .m_mem
-            .malloc_fcn
-            .expect("non-null function pointer")($size)
+    ($size:expr $(,)?) => {
+        alloc::alloc(malloc_layout($size as usize)) as *mut c_void
     };
 }
 macro_rules! REALLOC {
-    ($parser:path, $ptr:expr, $size:expr $(,)?) => {
-        (*$parser)
-            .m_mem
-            .realloc_fcn
-            .expect("non-null function pointer")($ptr, $size)
+    ($ptr:expr, $size:expr $(,)?) => {
+        alloc::realloc($ptr as *mut u8, malloc_layout($size as usize), $size as usize) as *mut c_void
     };
 }
 macro_rules! FREE {
-    ($parser:expr, $ptr:expr $(,)?) => {
-        (*$parser)
-            .m_mem
-            .free_fcn
-            .expect("non-null function pointer")($ptr)
+    ($ptr:expr $(,)?) => {
+        // FIXME: get the actual layout somehow
```

Comment:
Fun story: libstd doesn't care about the layout anyway, since it internally calls `free` with just the pointer.